python/tvm/relay/op/strategy/x86.py

```diff
@@ -347,12 +348,20 @@ def dense_strategy_cpu(attrs, inputs, out_type, target):
 def batch_matmul_strategy_cpu(attrs, inputs, out_type, target):
     """batch_matmul x86 strategy"""
     strategy = _op.OpStrategy()
-    strategy.add_implementation(
-        wrap_compute_batch_matmul(topi.x86.batch_matmul),
-        wrap_topi_schedule(topi.x86.schedule_batch_matmul),
-        name="batch_matmul.x86",
-        plevel=10,
-    )
+    if is_dynamic(out_type):
+        strategy.add_implementation(
```

Comment:
shouldn't this one be in generic.py?

Reply:
Hmm, something very similar to this is already in generic.py, what I'm trying to do here is short-circuit the schedule if we have dynamic shapes. The x86 schedule, as written, assumes static shapes and breaks during schedule construction if I give it a dynamic input. Is there a cleaner way to do that short-circuit in generic.py?

Reply:
I am not quite sure what is a better to do this though. @icemelon9 thoughts?

Reply:
What would the behavior be if instead we only had `if not is_dynamic(out_type)` to register the x86 schedule? I would think that the generic strategy would be used even if we dont readd it here.

Reply:
I'll give that a try! I'll report back shortly

Reply:
Unforunately, it seems like the compile engine can't find any schedules if I do this:

```
E             File "/home/mbrookhart/repos/mbrookhart_tvm/python/tvm/relay/backend/compile_engine.py", line 289, in lower_call
E               op, call.attrs, inputs, ret_type, target, use_autotvm=False
E             File "/home/mbrookhart/repos/mbrookhart_tvm/python/tvm/relay/backend/compile_engine.py", line 188, in select_implementation
E               best_plevel_impl = max(all_impls, key=lambda x: x.plevel)
E           ValueError: max() arg is an empty sequence
```